azkaban-common/src/main/java/azkaban/executor/container/watch/AzPodStatusExtractor.java

```diff
@@ -568,33 +571,43 @@ private void logAppContainerStatuses() {
    * @return
    */
   public AzPodStatus createAzPodStatus() {
+    // All pod statuses are checked for any status extracted from the pod event. If multiple
+    // statuses are detected, the first derived pod status will be returned, and the conflict
+    // will be logged
+    List<AzPodStatus> derivedStatuses = new ArrayList<>();
+
     if (checkForAzPodRequested()) {
-      return AzPodStatus.AZ_POD_REQUESTED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_REQUESTED);
     }
     logInitContainerStatuses();
     if (checkForAzPodScheduled()) {
-      return AzPodStatus.AZ_POD_SCHEDULED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_SCHEDULED);
     }
     logAppContainerStatuses();
     if (checkForAzPodInitContainersRunning()) {
-      return AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING;
+      derivedStatuses.add(AzPodStatus.AZ_POD_INIT_CONTAINERS_RUNNING);
     }
     if (checkForAzPodAppContainerStarting()) {
-      return AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING;
+      derivedStatuses.add(AzPodStatus.AZ_POD_APP_CONTAINERS_STARTING);
     }
     if (checkForAzPodReady()) {
-      return AzPodStatus.AZ_POD_READY;
+      derivedStatuses.add(AzPodStatus.AZ_POD_READY);
     }
     if (checkForAzPodCompleted()) {
-      return AzPodStatus.AZ_POD_COMPLETED;
+      derivedStatuses.add(AzPodStatus.AZ_POD_COMPLETED);
     }
     if (checkForAzPodInitFailure()) {
-      return AzPodStatus.AZ_POD_INIT_FAILURE;
+      derivedStatuses.add(AzPodStatus.AZ_POD_INIT_FAILURE);
     }
     if (checkForAzPodAppFailure()) {
-      return AzPodStatus.AZ_POD_APP_FAILURE;
+      derivedStatuses.add(AzPodStatus.AZ_POD_APP_FAILURE);
+    }
+    if (derivedStatuses.size() > 1) {
+      logger.warn(format("%d pod statuses are derived from current pod watch event: "
+          + derivedStatuses.stream().map(s -> s.toString()).collect(Collectors.joining(","))),
+          derivedStatuses.size());
```

Comment:
Recommend rewriting this as:
```
logger.warn(format("%d pod statuses are derived from current pod watch event: %s ",
          derivedStatuses.size(),
          derivedStatuses.stream().map(s -> s.toString()).collect(Collectors.joining(","))));
```
This avoids mixing the 2 styles of composing log message (format string versus string-concat) and makes a little easier to read the statement.